src/loaders/api/register-api.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2020-2025 New Relic, Inc. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { handle } from '../../common/event-emitter/handle'
+import { warn } from '../../common/util/console'
+import { isValidTarget } from '../../common/util/target'
+import { FEATURE_NAMES } from '../features/features'
+import { now } from '../../common/timing/now'
+import { SUPPORTABILITY_METRIC_CHANNEL } from '../../features/metrics/constants'
+
+/**
+ * @typedef {import('./register-api-types').RegisterAPI} RegisterAPI
+ */
+
+/**
+ * Builds the api object that will be returned from the register api method.
+ * Also conducts certain side-effects, such as harvesting a PageView event when triggered and gathering metadata for the registered entity.
+ * @param {Object} agentRef the reference to the base agent instance
+ * @param {Object} handlers the shared handlers to be used by both the base agent's API and the external target's API
+ * @param {Object} target the target information to be used by the external target's API to send data to the correct location
+ * @returns {RegisterAPI} the api object to be returned from the register api method
+ */
+export function buildRegisterApi (agentRef, handlers, target) {
+  const attrs = {}
+  warn(53, 'newrelic.register')
+  if (!isValidTarget(target, false)) {
+    const invalidTargetResponse = () => warn(47, target)
+    invalidTargetResponse()
+    return {
+      addPageAction: invalidTargetResponse,
+      log: invalidTargetResponse,
+      noticeError: invalidTargetResponse,
+      setApplicationVersion: invalidTargetResponse,
+      setCustomAttribute: invalidTargetResponse,
+      setUserId: invalidTargetResponse,
+      /** metadata */
+      metadata: {
+        customAttributes: attrs,
+        target
+      }
+    }
```

Comment:
I think you can just return undefined if the `target` is not valid. It's counterproductive to return a pseudo entity that just spams warning rather than user knowing right away something is wrong.

Reply:
I worry that risks potential to affect customer billing, since it will trigger JS errors to fire in the application if the APIs are undefined. This matches the behavior we use for setting the APIs in an empty state in the main agent interfaces as well